compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.test.cpp

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/RemoveUnnecessaryTransposeNetPass.h"
+
+#include <luci/IR/CircleNode.h>
+
+#include <luci/test/TestIOGraph.h>
+
+#include <gtest/gtest.h>
+
+namespace
+{
+
+using namespace luci::test;
+
+class TransposeReshapeTransposeGraph : public TestIOGraph
+{
+
+public:
+  void init(ShapeU32 in_shape, ShapeU32 front_perm, ShapeU32 mid_shape, ShapeU32 back_perm,
+            ShapeU32 out_shape)
+  {
+    TestIOGraph::init(in_shape, out_shape);
+
+    auto init_circle_const = [](luci::CircleConst *const_node, ShapeU32 shape) {
+      const_node->dtype(loco::DataType::S32);
+      const_node->size<loco::DataType::S32>(shape.size());
+      uint32_t i = 0;
+      for (auto v : shape)
+      {
+        const_node->at<loco::DataType::S32>(i++) = v;
+      }
+    };
+
+    _front_perm = g()->nodes()->create<luci::CircleConst>();
+    {
+      _front_perm->name("front_transpose/perm");
+      init_circle_const(_front_perm, front_perm);
+    }
+
+    _front_transpose = g()->nodes()->create<luci::CircleTranspose>();
+    {
+      _front_transpose->a(input());
+      _front_transpose->name("front_transpose");
+      _front_transpose->perm(_front_perm);
+    }
+
+    _mid_shape = g()->nodes()->create<luci::CircleConst>();
+    {
+      _mid_shape->name("mid_reshpae/shape");
+      init_circle_const(_mid_shape, mid_shape);
+    }
+
+    _mid_reshape = g()->nodes()->create<luci::CircleReshape>();
+    {
+      _mid_reshape->name("mid_reshape");
+      _mid_reshape->tensor(_front_transpose);
+      _mid_reshape->shape(_mid_shape);
+    }
+
+    _back_perm = g()->nodes()->create<luci::CircleConst>();
+    {
+      _back_perm->name("back_transpose/perm");
+      init_circle_const(_back_perm, back_perm);
+    }
+
+    _back_transpose = g()->nodes()->create<luci::CircleTranspose>();
+    {
+      _back_transpose->name("back_transpose");
+      _back_transpose->a(_mid_reshape);
+      _back_transpose->perm(_back_perm);
+    }
+
+    output()->from(_back_transpose);
+  }
+
+private:
+  luci::CircleTranspose *_front_transpose;
+  luci::CircleConst *_front_perm;
+
+  luci::CircleReshape *_mid_reshape;
+  luci::CircleConst *_mid_shape;
+
+  luci::CircleTranspose *_back_transpose;
+  luci::CircleConst *_back_perm;
+};
+
+} // namespace
+
+TEST(RemoveUnnecessaryTransposeNetPass, rank_reduction_pattern1)
```

Comment:
I made many tests for the previous draft. 
But now, its' target pattern become narrow, Some patterns are not targeted now. 
So, I disabled some tests. 

```
➜ ./build/compiler/luci/pass/luci_pass_test --gtest_filter='RemoveUnnecessaryTransposeNetPass.*'  
Running main() from /home/sseung/Workspace/ONE_second/externals/GTEST/googletest/src/gtest_main.cc
Note: Google Test filter = RemoveUnnecessaryTransposeNetPass.*
[==========] Running 7 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 7 tests from RemoveUnnecessaryTransposeNetPass
[ RUN      ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern1
[       OK ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern1 (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern2
[       OK ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern2 (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern3
[       OK ] RemoveUnnecessaryTransposeNetPass.rank_reduction_pattern3 (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.one_rank_pattern
[       OK ] RemoveUnnecessaryTransposeNetPass.one_rank_pattern (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.identity_pattern
[       OK ] RemoveUnnecessaryTransposeNetPass.identity_pattern (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.basic_pattern1_NEG
[       OK ] RemoveUnnecessaryTransposeNetPass.basic_pattern1_NEG (0 ms)
[ RUN      ] RemoveUnnecessaryTransposeNetPass.basic_pattern2_NEG
[       OK ] RemoveUnnecessaryTransposeNetPass.basic_pattern2_NEG (0 ms)
[----------] 7 tests from RemoveUnnecessaryTransposeNetPass (0 ms total)

[----------] Global test environment tear-down
[==========] 7 tests from 1 test suite ran. (0 ms total)
[  PASSED  ] 7 tests.

  YOU HAVE 6 DISABLED TESTS
```